test-records.js

```diff
@@ -112,6 +113,56 @@ validator('companies', cdb_schema, (json) => {
                 msg: "Record has `quality` of `tested` but doesn't specify `required-elements`. See: https://github.com/datenanfragen/data/issues/811",
             });
     }
+    // whitespace check
+    Object.keys(json).forEach((key) => {
+        if (typeof json[key] === 'string' && json[key] !== json[key].trim())
+            errors.push(`Seems like \`${key}\` isn't trimmed, i.e. it contains leading or trailing whitespace.`);
+    });
+
+    //// ADDRESS FORMATTING
+    const address_lines = json['address'].split('\n');
+    if (address_lines.length < 2) errors.push('`address` is not formatted with newlines (\\n).');
+    for (line of address_lines) {
+        if (line !== line.trim()) {
+            ("`address` isn't trimmed, i.e. contains unnecessary whitespace.");
+        }
+    }
+    if (address_lines[0].trim() === json['name']) errors.push(' Record includes `name` in the `address`.');
```

Comment:
I think we can generalize that to ``json['address'].includes(`\n${json['name']}\n`)`` (maybe even without the newlines?)—the name really shouldn't appear anywhere in the address.

Reply:
Sometimes companies have their own street named after them e.g. Airbus-Allee in Bremen. That means that we can't ditch the newlines.

Reply:
Wouldn't `name` then be `Airbus Somethingorother SA`? But yeah, we can keep them just in case. These tests should be hard fails, not just hints.

But in that case I would like to revise my suggestion. :D `address_lines.includes(json['name'])` is a lot cleaner.